perfkitbenchmarker/scripts/object_storage_api_tests.py

```diff
@@ -271,81 +262,238 @@ def PercentileCalculator(numbers):
   return result
 
 
-def _ListObjects(storage_schema, bucket, prefix, host_to_connect=None):
-  """List objects under a bucket given a prefix.
+# ### Backends ###
 
-  Args:
-    storage_schema: The address schema identifying a storage. e.g., "gs"
-    bucket: Name of the bucket.
-    prefix: A prefix to list from.
-    host_to_connect: An optional endpoint string to connect to.
+class ObjectStorageBackend(object):
```

Comment:
I see why putting them under one directory makes it easier, but my ideal vision is that the benchmark doesn't have to copy over all of the provider scripts, only the one that's actually going to be used. Here's my proposed final organization, after this and 1-2 more PRs:

  - Each provider directory has a `scripts/` subdirectory, containing their `ObjectStorageApiProvider`.
  - Each provider directory also has a new resource called `ObjectStorageService`, which encapsulates the CLI commands that live in the overall benchmarking script right now. It also does two more things: provides a list of packages that need to be installed on a VM to use that provider's object storage API (right now I think we spend a lot of our overall provisioning time installing Azure's client library on VMs where we're not going to use it), and provides a list of files that need to be copied to a VM to provide the `ObjectStorageApiProvider` object.

Then the benchmarking script can install only the packages that are actually needed, and copy over only the provider object that is actually needed. How does this end-state sound to you?